File: back-end/src/main/java/net/easyfinance/core/config/MvcConfig.java
package net.easyfinance.core.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class MvcConfig implements WebMvcConfigurer {

    @Value("${view.frontend-managed-urls}")
    private String[] frontendManagedUrls;

    @Value("${view.forward-to-frontend-url}")
    private String forwardToFrontendUrl;

    @Override
    public void addResourceHandlers(ResourceHandlerRegistry registry) {
        registry.addResourceHandler("/**")
                .addResourceLocations("classpath:/resources/");
    }

    @Override
    public void addViewControllers(ViewControllerRegistry registry) {
        for (String path : frontendManagedUrls) {
            registry.addViewController(path).setViewName(forwardToFrontendUrl);
        }
    }
}

File: back-end/src/main/kotlin/net/easyfinance/core/config/MvcConfig.kt
package net.easyfinance.core.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
open class MvcConfig : WebMvcConfigurer {

    @Value("\${view.frontend-managed-urls}")
    private lateinit var frontendManagedUrls: Array<String>

    @Value("\${view.forward-to-frontend-url}")
    private lateinit var forwardToFrontendUrl: String

    override fun addResourceHandlers(registry: ResourceHandlerRegistry) {
        super.addResourceHandlers(registry)
        registry.addResourceHandler("/**")
                .addResourceLocations("classpath:/", "classpath:/META-INF/resources/")
    }

    override fun addViewControllers(registry: ViewControllerRegistry) {
        frontendManagedUrls.forEach { registry.addViewController(it).setViewName(forwardToFrontendUrl) }
    }
}